api/submission/submit.go

```diff
@@ -10,13 +10,15 @@ import (
 type SubmitRequest struct {
     core.APIRequestAssignmentContext
     core.MinRoleStudent
+    LateAcknowledgment bool `json:"late-acknowledgment"`
     Files core.POSTFiles
 
     Message string `json:"message"`
 }
 
 type SubmitResponse struct {
     Rejected bool `json:"rejected"`
+    RequireLateAcknowledgment bool `json:"require-late-acknowledgment"`
```

Comment:
I think we can generalize this fairly cheaply.
Instead of a field just for late rejections, you can all a `RejectReason` string field that pulls from an enum.